src/plugins/wizard/public/application/utils/get_top_nav_config.tsx

```diff
@@ -50,8 +50,20 @@ interface TopNavConfigParams {
 
 export const getTopNavConfig = (
   { visualizationIdFromUrl, savedWizardVis, saveDisabledReason, dispatch }: TopNavConfigParams,
-  { history, toastNotifications, i18n: { Context: I18nContext } }: WizardServices
+  services: WizardServices
 ) => {
+  const {
+    i18n: { Context: I18nContext },
+    embeddable,
+    scopedHistory,
+  } = services;
+
+  const { originatingApp: originatingApp } =
+    embeddable
+      .getStateTransfer(scopedHistory)
+      .getIncomingEditorState({ keysToRemoveAfterFetch: ['id', 'input'] }) || {};
+  const stateTransfer = embeddable.getStateTransfer();
```

Comment:
Are you talking about the `getAppNameFromId = {stateTransfer.getAppNameFromId}` that we pass in as a prop in `<SavedObjectSaveModalOrigin>`? I am following the pattern of how visualization plugin does it. It seems like for getAppNameFromId prop, we are getting that function definition from embeddable.getStateTransfer(), then we call that the getAppNameFromId function with originatingApp passed in to actually get the originating app name. 
```
const origin = props.getAppNameFromId
      ? props.getAppNameFromId(props.originatingApp) || props.originatingApp
      : props.originatingApp;
```

Reply:
Mostly I was just curious to learn more about `embeddable.getStateTransfer` and the distinction between when it's called with an argument and when it's called without, since we do it both ways in this block.

Reply:
```
getStateTransfer: (history?: ScopedHistory) => {
        return history
          ? new EmbeddableStateTransfer(core.application.navigateToApp, history, this.appList)
          : this.outgoingOnlyStateTransfer;
      },
```

It can be called with a ScopedHistory or without a ScopedHistory, and both situations it will return a `new EmbeddableStateTransfer`. 
If we have a scopedhistory, it will return `new EmbeddableStateTransfer(core.application.navigateToApp, history, this.appList)`; if not, it will return 
`new EmbeddableStateTransfer(core.application.navigateToApp, undefined, this.appList )`. Within the EmbeddableStateTransfer class, scopedHistory is only used in functions such as `getIncomingState`, `navigateToWithState` etc that actually require the information from the scoped history's location state; `getAppNameFromId` does not require scopedHistory since it only requires the id of the app to fetch the app name for.